packages/cubejs-backend-native/src/transport.rs

```diff
@@ -57,17 +57,34 @@ impl TransportService for NodeBridgeTransport {
             },
             user: Some(ctx.access_token.clone()),
         })?;
-        let response = call_js_with_channel_as_callback::<V1MetaResponse>(
+
+        let response: serde_json::Value = call_js_with_channel_as_callback(
             self.channel.clone(),
             self.on_meta.clone(),
             Some(extra),
         )
         .await?;
+
         trace!("[transport] Meta <- {:?}", response);
 
-        Ok(MetaContext {
-            cubes: response.cubes.unwrap_or_default(),
-        })
+        let meta_response = match serde_json::from_value::<V1MetaResponse>(response.clone()) {
+            Ok(r) => {
+                // If `r.cubes` is `None`, then assume response includes an error
+                let cubes = r.cubes.unwrap_or_default();
```

Comment:
I am thinking about changing Open Spec for our API to make `cubes` as a required field to skip this check. WDYT?